pkg/testcase/service.go

```diff
@@ -88,3 +90,43 @@ func TestServiceLoadBalancer(deleteWorkload bool) {
 		Expect(err).NotTo(HaveOccurred(), "Loadbalancer manifest not deleted")
 	}
 }
+
+func TestServiceNodePortDualStack(testinfo map[string]string) {
+	nodeExternalIP := shared.FetchNodeExternalIP()
+	nodeport, err := shared.FetchServiceNodePort(testinfo["namespace"], testinfo["svc"])
+	Expect(err).NotTo(HaveOccurred(), err)
+	
+	for _, ip := range nodeExternalIP {
+		if strings.Contains(ip,":") {
+			ip = shared.IPv6IPEncloser(ip)
+		}
+		err = assert.CheckComponentCmdNode(
+			"curl -sL --insecure http://"+ip+":"+nodeport+"/name.html",
+			testinfo["expected"],
+			shared.BastionIP)
+		Expect(err).NotTo(HaveOccurred(), err)
+	}
+}
+
+func TestServiceClusterIPDualStack(testInfo map[string]string) {		
+	res, port, _ := shared.FetchClusterIPs(testInfo["namespace"], testInfo["svc"])
+	clusterIPs := strings.Split(res, " ")
+	nodeExternalIPs := shared.FetchNodeExternalIP()
+	
+	for _, clusterIP := range clusterIPs {
+		if strings.Contains(clusterIP,":") {
+			clusterIP = shared.IPv6IPEncloser(clusterIP)
+		}
+		err := assert.ValidateOnNode(nodeExternalIPs[0], 
+			"curl -sL --insecure http://"+ clusterIP +":"+ port, testInfo["expected"])
+		Expect(err).NotTo(HaveOccurred(), err)
+	}
+}
+
+func TestServiceSpecContainsSubstring(svc, expected string) {
+	cmd := "kubectl get svc "+svc+" -n default"+
```

Comment:
also should not be living in the test file, and its not a test actually right.

Reply:
This is kind of asserting and contributes to the test... can move it to assert

Reply:
Updated